russh-util/src/runtime.rs

```diff
@@ -0,0 +1,110 @@
+#[cfg(not(target_arch = "wasm32"))]
+pub use native::*;
+#[cfg(target_arch = "wasm32")]
+pub use wasm::*;
+
+#[derive(Debug)]
+pub struct JoinError {
+    #[cfg(not(target_arch = "wasm32"))]
+    inner: tokio::task::JoinError,
```

Comment:
I see that the current approach with `JoinHandle` seems to align with your preference. However, I’d like to propose a discussion on possibly using a one-shot channel as a unified simulation for the handle across targets. This would significantly reduce the code complexity and maintain a consistent type between WASM and non-WASM environments. By doing so, we could simplify the overall implementation while still achieving the desired functionality.